### ISSUE TITLE
Scene Summary & Dynamic Description Testing

### ISSUE THREAD

**Issue by terracoda** [phetsims/balloons-and-static-electricity#371]
Testing Date: March 28, 2018
PhET Test 15:45 NT/ 12:15 MT
Environment: Ally View in Safari, Mac OS, no screen reader.

Note that for object dynamic descriptions the [BASE Charge Views Design Doc](https://docs.google.com/document/d/1EOeMXgu0Cvqu1lh9WgFp_21enK8Xdy5EI_H1l6tBsvA/edit#) has clear examples laid out in tables for all charge views. I don't think I have the same thing for he scene summary (sorry).

**Scene Summary Page Load** 
- Show all charges - good - including balloon special case. Nice!
- Show no charges - good
- [x] Show charge differences - balloon description missing object's net charge description
![screen shot 2018-03-28 at 15 57 29](https://user-images.githubusercontent.com/1902116/38048745-c981bd4e-32a0-11e8-9f23-d15b8c07c919.png)
Yellow balloon SS description should end with "showing no charges" and not "no".
- [x] Same issue with two balloons are present. "Each balloon has zero net charge, no." should be "Each balloon has zero net charge, showing no charges."

**Scene Summary Balloon or Balloons at Wall** 
- Show all charges 
  - [x] Induced charge description has extra instance of "Positive charges do not move."
![ss-all-charges-induced-charge](https://user-images.githubusercontent.com/1902116/38050748-a6e7cd72-32a6-11e8-91c2-89fece53b533.png)
 - [x] Similar problem with 2 balloons at wall:
![ss-two-balloons-at-wall](https://user-images.githubusercontent.com/1902116/38050819-db2cc696-32a6-11e8-827d-96bdd310d0c2.png)

**Dynamic Descriptions on Page Load**
- Show all charges 
  - Sweater and Wall are good. 
  - [x] Yellow balloon missing visual object charge description
![yb](https://user-images.githubusercontent.com/1902116/38049005-90e97692-32a1-11e8-92b7-9438d40c7004.png)
 - [x] Same issue for both balloons when second balloon is added. 
 - [x] Same issue when balloon or balloons have charge.
    - [x] Charge description is missing: "more negative charges than positive charges"

- Show no charges 
  - Sweater and Wall, good. 
  - [x] Yellow balloon has extra period after location (This might be an empty string?)
![yb-show-no-charges](https://user-images.githubusercontent.com/1902116/38049130-eef99064-32a1-11e8-8b43-2392a1365df9.png)
 - [x]  Same extra period when both balloons are present.
 - [x] Same extra period after interaction and balloon has charge.

- Show charge differences 
  - Wall is good.
  - [x] Sweater is missing final period (on page load and after interaction)
  - [x] Yellow balloon missing visual net charge description
![s-yb-diffs](https://user-images.githubusercontent.com/1902116/38049406-c299ea7c-32a2-11e8-82de-3c9598588b42.png)
Yellow balloon needs "showing no charges." instead of the the "no."
- [x] Same missing description for both balloons when both balloons are present.
- [x] Same missing description with negatively charged balloon or balloons.

**Special Case Sweater object description when sweater is depleted of negative charges**
- [x] Show all charges - correct except for missing period at end.
- Show no charges - correct
- [x] Show charge differences - incorrect description
![sweater-incorrect-description](https://user-images.githubusercontent.com/1902116/38051424-d0479024-32a8-11e8-8300-dbd16d47aff2.png)
Need one more parameter, "all" for Show charge differences view: "At left edge of Play Area. Has {{positive}} net charge, showing {{all}} positive charges."
- [x] And I think the same strings (without location) can be used in the Scene Summary as well: "Sweater has {{positive}} net charge, showing {{all}} positive charges."

**Wall description with 2 charged balloons needs work**
- [x] I am assuming this description is still being implemented. I think it is covered in the [Two-balloon design doc here](https://docs.google.com/document/d/1LrOWT4RZiS4emrumW479POK--zWveWiylZMd7unNyi0/edit#heading=h.r74ohrg0uxpq)
![wall-two-balloons](https://user-images.githubusercontent.com/1902116/38052024-860fdc44-32aa-11e8-861d-f2f1a828234a.png)

**Comment by terracoda** [comment]
@jessegreenberg, these are minor corrections. I will continue to test the dynamic descriptions, including all charge views. I'll keep them in this issue for now, and can move into indy issues if you prefer.

**Comment by jessegreenberg** [comment]
Like #372 this may be stale, I will check.

**Comment by terracoda** [comment]
@jessegreenberg, sorry, I tested an old version. Shall we just close this issue, and I can test again.

**Comment by terracoda** [comment]
@jessegreenberg, I'll just test again and see if these issues have been addressed. No need to check.

**Comment by terracoda** [comment]
@jessegreenberg, I have deleted the issues, and I am retesting on PhET Test.

**Comment by terracoda** [comment]
@jessegreenberg, I have re-tested the Scene Summary and Dynamic Descriptions using PhET Test. I tried to be thorough. If you need me to do this kind of thing differently, let's discuss.

**Comment by terracoda** [comment]
Sorry @jessegreenberg, I forgot to re-assign to you yesterday.

**Comment by jessegreenberg** [comment]
Thanks @terracoda I  added checkboxes next to the items ini your list that need work.

**Comment by jessegreenberg** [comment]
Much of these bugs are caused by changes made for https://github.com/phetsims/balloons-and-static-electricity/issues/341, sorry @terracoda.

**Comment by jessegreenberg** [comment]
I re implemented that sweater description, all related issues should be fixed.

**Comment by jessegreenberg** [comment]
Thanks for testing @terracoda, all in this issue should be fixed and up to date. Closing.
